modules/ref_device_module/include/ref_device_module/ref_device_impl.h

```diff
@@ -36,7 +36,7 @@ class RefDeviceImpl final : public Device
     static DeviceTypePtr CreateType();
 
     // IDevice
-    DeviceInfoPtr onGetInfo() override;
+    void onSetDeviceInfo() override;
```

Comment:
As mentioned above, you should not be able to set the device info externally like that. Additionally, this is a breaking change for any current device implementation - let's try to avoid those at the moment, and do them only where necessary.